Date: Fri, 27 Oct 2017 17:59:18 +0000
Subject: PM / hibernate: update the resume offset on SNAPSHOT_SET_SWAP_AREA

The SNAPSHOT_SET_SWAP_AREA is supposed to be used to set the hibernation
offset on a running kernel to enable hibernating to a swap file.
However, it doesn't actually update the swsusp_resume_block variable. As
a result, the hibernation fails at the last step (after all the data is
written out) in the validation of the swap signature in
mark_swapfiles().

Before this patch, the command line processing was the only place where
swsusp_resume_block was set.

Signed-off-by: Aleksei Besogonov <cyberax@amazon.com>
Signed-off-by: Munehisa Kamata <kamatam@amazon.com>
Signed-off-by: Anchal Agarwal <anchalag@amazon.com>
Reviewed-by: Munehisa Kamata <kamatam@amazon.com>
Reviewed-by: Eduardo Valentin <eduval@amazon.com>
Signed-off-by: Samuel Mendoza-Jonas <samjonas@amazon.com>

diff --git a/kernel/power/user.c b/kernel/power/user.c
--- a/kernel/power/user.c
+++ b/kernel/power/user.c
@@ -243,6 +243,10 @@ static int snapshot_set_swap_area(struct snapshot_data *data,
 	if (data->swap < 0)
 		return swdev ? -ENODEV : -EINVAL;
 	data->dev = swdev;
+
+	swsusp_resume_device = swdev;
+	swsusp_resume_block = offset;
+
 	return 0;
 }
 
